releases/release-5.1.0.md

```diff
@@ -0,0 +1,345 @@
+    - Fix multiple `Region is Unavailable` issues that might occur in frequent scheduling situations
+    - Fix `Region is Unavailable` issue that might occur in some high stress write situations
+
+- Avoid frequently reading the `mysql.stats_histograms` table if the cached statistics is up-to-date to avoid high CPU usage [#24317](https://github.com/pingcap/tidb/pull/24317)
+
++ TiKV
+
+    - Use `zstd` to compress Region snapshots, preventing large space differences between nodes in case of heavy scheduling or scaling [#10005](https://github.com/tikv/tikv/pull/10005)
+    - Solve OOM issues in multiple cases [#10183](https://github.com/tikv/tikv/issues/10183)
+
+        - Add memory usage tracking for each module
+        - Solve the OOM issue caused by oversized Raft entries cache
+        - Solve the OOM issue caused by stacked GC tasks
+        - Solve the OOM issue caused by fetching too many Raft entries from the Raft log to memory at one time
+
+    - Split Regions more evenly to mitigate the issue that the growth of Region size exceeds the splitting speed when there are hotspot writes [#9785](https://github.com/tikv/tikv/issues/9785)
+
++ TiFlash
+
+    - Support `Union All`, `TopN`, and `Limit` functions
+    - Support the Cartesian product including left outer join and semi anti join in MPP mode
+    - Optimize lock operations to avoid that running DDL statements and read operations are blocked by each other
+    - Optimize cleanup of expired data by TiFlash 
+    - Support further filtering of query filters on `timestamp` columns at the TiFlash storage level
+    - Improve the startup and scalability speed of TiFlash when a large number of tables are in a cluster
+    - Improve TiFlash compatibility when running on unknown CPUs
+
++ PD
+    - Avoid unexpected statistics after adding the `scatter region` scheduler [#3602](https://github.com/pingcap/pd/pull/3602)
+    - Solve multiple scheduling issues in the scaling process
+
+        - Optimize the generation process of replica snapshots to solve slow scheduling issues during scaling: 
+[#3563](https://github.com/tikv/pd/issues/3563) [#10059](https://github.com/tikv/tikv/pull/10059) [#10001](https://github.com/tikv/tikv/pull/10001)
+        - Solve slow scheduling issues caused by heartbeat pressure due to traffic changes 
+[#3693](https://github.com/tikv/pd/issues/3693) [#3739](https://github.com/tikv/pd/issues/3739) [#3728] (https://github.com/tikv/pd/issues/3728) [#3751](https://github.com/tikv/pd/issues/3751)
+        - Reduce the space discrepancies of large clusters due to scheduling, and optimize the scheduling formula to prevent the bursting issue (which is similar to heterogeneous space clusters) caused by large compression rate discrepancies    [#3592](https://github.com/tikv/pd/issues/3592) [#10005](https://github.com) /tikv/tikv/pull/10005)
```

Comment:
```suggestion
        - Reduce the space discrepancies of large clusters due to scheduling, and optimize the scheduling formula to prevent the bursting issue (which is similar to heterogeneous space clusters) caused by large compression rate discrepancies    [#3592](https://github.com/tikv/pd/issues/3592) [#10005](https://github.com/tikv/tikv/pull/10005)
```